atox/src/main/kotlin/ui/AvatarImageView.kt

```diff
@@ -0,0 +1,100 @@
+package ltd.evilcorp.atox.ui
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.FrameLayout
+import androidx.constraintlayout.widget.ConstraintLayout
+import kotlin.math.pow
+import kotlin.math.sqrt
+import ltd.evilcorp.atox.R
+import ltd.evilcorp.atox.databinding.AvatarImageLayoutBinding
+
+class AvatarImageView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
+    FrameLayout(context, attrs, defStyleAttr) {
+
+    companion object {
+        private const val STATUS_INDICATOR_SIZE_RATIO_WITH_AVATAR = 12f / 50
+    }
+
+    private var _binding: AvatarImageLayoutBinding? = null
+    private val binding get() = _binding!!
+
+    val avatarImage get() = binding.avatarImage
+
+    val statusIndicator get() = binding.statusIndicator
+
+    var size: Int = 0
+        set(newSize) = binding.run {
+            avatarImage.layoutParams.width = newSize
+            avatarImage.layoutParams.height = newSize
+
+            // Calculating status indicator size and margin in relation to the overall size
+            val newSizeFloat = newSize.toFloat()
+            val statusIndicatorSize = newSizeFloat * STATUS_INDICATOR_SIZE_RATIO_WITH_AVATAR
+            val avatarImageRadius = newSizeFloat / 2
+            val avatarImageDiagonal = sqrt(newSizeFloat.pow(2) + newSizeFloat.pow(2)) // Pythagorean theorem
+            val avatarImageDistanceFromCorner = avatarImageDiagonal / 2 - avatarImageRadius
+            val statusIndicatorDiagonal =
+                sqrt(statusIndicatorSize.pow(2) + statusIndicatorSize.pow(2)) // Pythagorean theorem
+            val statusIndicatorMarginDiagonal = (avatarImageDistanceFromCorner - statusIndicatorDiagonal / 2)
+            val statusIndicatorMargin =
+                sqrt(statusIndicatorMarginDiagonal.pow(2) / 2).toInt() // Pythagorean theorem
+
+            val statusIndicatorLayoutParams = ConstraintLayout.LayoutParams(
+                ConstraintLayout.LayoutParams.WRAP_CONTENT,
+                ConstraintLayout.LayoutParams.WRAP_CONTENT
+            )
+            statusIndicatorLayoutParams.bottomToBottom = R.id.parent
```

Comment:
If you set all of these on the `statusIndicatorLayoutParams` in a `.apply` block you don't have to repeat the name 6 times. :P You could even just do
```
statusIndicator.layoutParams = ConstraintLayout.LayoutParams(
    ConstraintLayout.LayoutParams.WRAP_CONTENT,
    ConstraintLayout.LayoutParams.WRAP_CONTENT
).apply {
    bottomToBottom = R.id.parent
    endToEnd = R.id.parent
    // ...
}
```

Reply:
Forgot about that nice Kotlin's feature!